File: solutions/cpp/reverse-string/1/reverse_string.cpp
#include "reverse_string.h"

namespace reverse_string { 
    std::string reverse_string(std::string input) {
        std::string res(input.size(), '\0');
        for (int i = 0, j = input.size()-1; i <= j ; ++i, --j) {
            res[i] = input[j];
            res[j] = input[i];
        }
        return res;
    }
}

File: solutions/go/reverse-string/1/reverse_string.go
package reverse

// import "golang.org/x/text/unicode/norm"

func Reverse(input string) string {
	// input = norm.NFC.String(input)
    runes := []rune(input)
    for i, j := 0, len(runes) - 1; i < j; i, j = i+1, j-1 {
        runes[i], runes[j] = runes[j], runes[i]
    }
    return string(runes)
}
